mythril/analysis/modules/integer.py

```diff
@@ -270,11 +270,43 @@ def _search_children(statespace, node, expression, index=0, depth=0, max_depth=6
         current_state = node.states[j]
         current_instruction = current_state.get_current_instruction()
         if current_instruction['opcode'] in ('JUMPI', 'SSTORE'):
```

Comment:
For some reason, in case of BECToken.sol it never hapens that we find an SSTORE instruction. Therefore the following is not detected anymore:
```
    uint256 amount = uint256(cnt) * _value;
    require(cnt > 0 && cnt <= 20);
    require(_value > 0 && balances[msg.sender] >= amount);

    balances[msg.sender] = balances[msg.sender].sub(amount);
```

the weird thing is that it does work for the following:
```
contract Over {

  mapping(address => uint) balances;
  uint public totalSupply;

  function Token(uint _initialSupply) {
    balances[msg.sender] = totalSupply = _initialSupply;
  }

  function sendeth(address _to, uint _value) public returns (bool) {
    require(balances[msg.sender] - _value >= 0);
    balances[msg.sender] -= _value;
    if ( true) {
        balances[_to] += _value;
    }
    balances[_to] = add(2, _value);
    return true;
  }

  function add(uint256 a, uint256 b) internal constant returns (uint256) {
    uint256 c = a + b;
    assert(c >= a);
    return c;
  }

  function balanceOf(address _owner) public constant returns (uint balance) {
    return balances[_owner];
  }
}
```

Reply:
Weird indeed. The SSTORE should happen shortly after the MUL.